Apps/W1/APIV2/test/src/APIV2FALocationsE2E.Codeunit.al

```diff
@@ -0,0 +1,161 @@
+        // [SCENARIO] User can modify name in a Fixed Asset Location through a PATCH request.
+        Initialize();
+
+        // [GIVEN] A Fixed Asset Location exists with a name.
+        CreateFALocation(FALocation);
+        TempFALocation.TransferFields(FALocation);
+        FALocation.Code := LibraryUtility.GenerateRandomCodeWithLength(FALocation.FieldNo(Code), Database::"FA Location", 10);
+        TempFALocation.Name := LibraryUtility.GenerateGUID();
+        RequestBody := GetFALocationJSON(TempFALocation);
+
+        // [WHEN] The user makes a patch request to the service and specifies name field.
+        TargetURL := LibraryGraphMgt.CreateTargetURL(FALocation.SystemId, Page::"APIV2 - FA Locations", ServiceNameTxt);
+        LibraryGraphMgt.PatchToWebService(TargetURL, RequestBody, Response);
+
+        // [THEN] The response contains the new values.
+        VerifyProperties(Response, TempFALocation);
+
+        // [THEN] The Fixed Asset Location in the database contains the updated value.
+        FALocation.Get(FALocation.Code);
+        Assert.AreEqual(FALocation.Name, TempFALocation.Name, 'Names should be equal.');
+    end;
+
+    [Test]
+    procedure TestDeleteFALocation()
+    var
+        FALocation: Record "FA Location";
+        FALocationCode: Code[10];
+        TargetURL: Text;
+        Response: Text;
+    begin
+        // [SCENARIO] User can delete a Fixed Asset Location by making a DELETE request.
+        Initialize();
+
+        // [GIVEN] A Fixed Asset Location exists.
+        CreateFALocation(FALocation);
+        FALocationCode := FALocation.Code;
+
+        // [WHEN] The user makes a DELETE request to the endpoint for the Fixed Asset Location.
+        TargetURL := LibraryGraphMgt.CreateTargetURL(FALocation.SystemId, Page::"APIV2 - FA Locations", ServiceNameTxt);
+        LibraryGraphMgt.DeleteFromWebService(TargetURL, '', Response);
+
+        // [THEN] The response is empty.
+        Assert.AreEqual('', Response, 'DELETE response should be empty.');
+
+        // [THEN] The Fixed Asset Location is no longer in the database.
+        FALocation.SetRange(Code, FALocationCode);
+        Assert.IsTrue(FALocation.IsEmpty(), 'Fixed Asset Location should be deleted.');
+    end;
+
+    local procedure CreateFALocation(var FALocation: Record "FA Location")
+    begin
+        FALocation.Init();
+        FALocation.Code := LibraryUtility.GenerateRandomCodeWithLength(FALocation.FieldNo(Code), Database::"FA Location", 10);
+        FALocation.Name := LibraryUtility.GenerateGUID();
+        FALocation.Insert(true);
+        Commit();
+    end;
+
+    local procedure GetFALocationJSON(var FALocation: Record "FA Location") FALocationJSON: Text
+    begin
+        if FALocation.Code = '' then
+            FALocation.Code := LibraryUtility.GenerateRandomCodeWithLength(FALocation.FieldNo(Code), Database::"FA Location", 10);
+        if FALocation.Name = '' then
+            FALocation.Name := LibraryUtility.GenerateGUID();
```

Comment:
Is there a situation when these fields can be empty? As far as I can see, this procedure is always called for records with all fields initialized. I believe the intention of the method would be clearer without these conditions.

Reply:
The same story as above - in Locations there is also no way to have it empty - this is a copy from standard tests so for the reasons of having the same thing the same way I would leave as is. Bacasue then the change should be global for all test codeunits in this app.

![image](https://github.com/microsoft/ALAppExtensions/assets/106300783/303445c5-38ee-401f-8278-7a7cec3b314e)